app/submitter/submitter.py

```diff
@@ -141,3 +143,31 @@ def send_message(self, message, tx_id, questionnaire_id, case_id=None):
         finally:
             if connection:
                 self._disconnect(connection)
+
+
+class GCSFeedbackSubmitter:
+    def __init__(self, bucket_name):
+        client = storage.Client()
+        self.bucket = client.get_bucket(bucket_name)
+
+    def upload(self, metadata, payload):
+        blob = self.bucket.blob(str(uuid4()))
+        blob.metadata = metadata
+        blob.upload_from_string(
+            str({**payload, **metadata}).encode("utf8"), content_type="application/json"
+        )
```

Comment:
I think either payload should be updated in place since metadata is a subset of payload or assign a new variable for clarity instead of unpacking 2 dicts into a new dict.
```suggestion
        payload.update(metadata)
        blob = self.bucket.blob(str(uuid4()))
        blob.metadata = metadata
        blob.upload_from_string(
            str(payload).encode("utf8"), content_type="application/json"
        )
```

Reply:
I don't think this is needed